src/test/java/com/metamx/common/parsers/RegexParserTest.java

```diff
@@ -0,0 +1,189 @@
+    builder.put("HTTP status", "200");
+    builder.put("Error Code", "-");
+    builder.put("Bytes Sent", "113");
+    builder.put("Object Size", "-");
+    builder.put("Total Time", "7");
+    builder.put("Turn-Around Time", "-");
+    builder.put("Referrer", "-");
+    builder.put("User-Agent", "S3Console/0.4");
+    builder.put("Version ID", "-");
+
+    Assert.assertEquals(
+        "result",
+        builder.build(),
+        parsed
+    );
+  }
+
+  @Test
+  public void testAWSLogWithCrazyUserAgent()
+  {
+    final String pattern = "^([0-9a-f]+) ([\\w.-]+) \\[([\\w\\/: +-]+)\\] ([\\d.]+) ([^\\s]+) ([\\w]+) ([\\w.-]+) ([^\\s\"]+) \"([^\"]*)\" ([\\d-]+) ([\\w-]+) ([\\d-]+) ([\\d-]+) ([\\d-]+) ([\\d-]+) \"(.+)\" \"(.+)\" ([\\w-]+)$";
+
+    final List<String> fieldNames = Arrays.asList(
+        "Bucket Owner",
+        "Bucket",
+        "Time",
+        "Remote IP",
+        "Requester",
+        "Request ID",
+        "Operation",
+        "Key",
+        "Request-URI",
+        "HTTP status",
+        "Error Code",
+        "Bytes Sent",
+        "Object Size",
+        "Total Time",
+        "Turn-Around Time",
+        "Referrer",
+        "User-Agent",
+        "Version ID"
+    );
+
+    final Parser<String, Object> parser = new RegexParser(
+        pattern,
+        Optional.<String>absent(),
+        fieldNames
+    );
+    String data = "79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be mybucket [06/Feb/2014:00:01:00 +0000] 192.0.2.3 79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be 7B4A0FABBEXAMPLE REST.GET.VERSIONING - \"GET /mybucket?versioning HTTP/1.1\" 200 - 139 139 27 26 \"-\" \"() { foo;};echo; /bin/bash -c \"expr 299663299665 / 3; echo 333:; uname -a; echo 333:; id;\"\" -";
+
+    final Map<String, Object> parsed = parser.parse(data);
+    ImmutableMap.Builder builder = ImmutableMap.builder();
+    builder.put("Bucket Owner", "79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be");
+    builder.put("Bucket", "mybucket");
+    builder.put("Time", "06/Feb/2014:00:01:00 +0000");
+    builder.put("Remote IP", "192.0.2.3");
+    builder.put("Requester", "79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be");
+    builder.put("Request ID", "7B4A0FABBEXAMPLE");
+    builder.put("Operation", "REST.GET.VERSIONING");
+    builder.put("Key", "-");
+    builder.put("Request-URI", "GET /mybucket?versioning HTTP/1.1");
+    builder.put("HTTP status", "200");
+    builder.put("Error Code", "-");
+    builder.put("Bytes Sent", "139");
+    builder.put("Object Size", "139");
+    builder.put("Total Time", "27");
+    builder.put("Turn-Around Time", "26");
+    builder.put("Referrer", "-");
+    builder.put(
+        "User-Agent",
+        "() { foo;};echo; /bin/bash -c \"expr 299663299665 / 3; echo 333:; uname -a; echo 333:; id;\""
+    );
+    builder.put("Version ID", "-");
+
+    Assert.assertEquals(
+        "result",
+        builder.build(),
+        parsed
+    );
+  }
```

Comment:
if user agent is the only thing different between 2 then it might be useful to have a method
```
private void testAWS(String userAgent) {..}
```
and two test methods that call above with appropriate userAgent, that would reduce all duplication and be more clear.